### PR TITLE
DB-6478 Add result columns SUBMITTED, ENGINE and JOBTYPE in SYSCS_GET_RUNNING_OPERATIONS / LOCAL (2.5)

### PR DESCRIPTION
**Pull request by changli6**

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/iapi/sql/execute/SpliceOperation.java [hunk 1]
```diff
@@ -389,4 +389,13 @@
     String getVTIFileName();
 
     boolean accessExternalTable();
+
+    /**
+     * DB-6478
+     * Return the Query's Submit Time and Engine information
+     * for procedure SYSCS_UTIL.SYSCS_GET_RUNNING_OPERATIONS and
+     *    SYSCS_UTIL.SYSCS_GET_RUNNING_OPERATIONS_LOCAL
```
Comment (jleach4) [inline splice_machine/src/main/java/com/splicemachine/derby/iapi/sql/execute/SpliceOperation.java]: JavaDoc each field separately.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java [hunk 2]
```diff
@@ -398,6 +404,8 @@ public String getScopeName() {
 
     public void openCore(DataSetProcessor dsp) throws StandardException{
         try {
+            submittedTime = new SimpleDateFormat(timeStampFormat).format(new Date());
```
Comment (jleach4) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java]: expensive...  Not on open core please.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java [hunk 3]
```diff
@@ -459,16 +469,21 @@ private void openDistributed() throws StandardException{
     @Override
     public void openCore() throws StandardException{
         try {
+            submittedTime = new SimpleDateFormat(timeStampFormat).format(new Date());
```
Comment (jleach4) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java]: expensive.

**Comment by dgomezferro** [comment]
I would store the submission time and engine used in the OperationManager. Right now it has a Map from UUID to Pair<SpliceOperation, Thread>>, make it a Map from UUID to a value class OperationInfo (or some other name) that stores the SpliceOperation, Thread, submission time, engine and any other useful information.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java [hunk 4]
```diff
@@ -410,12 +418,14 @@ public void openCore(DataSetProcessor dsp) throws StandardException{
 
             activation.getLanguageConnectionContext().setControlExecutionLimiter(EngineDriver.driver().processorFactory().getControlExecutionLimiter(activation));
             returnedRows = false;
+            engineName = (dsp.getType() == DataSetProcessor.Type.SPARK) ? "SPARK" : "CONTROL";
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java]: Make the enum return it's name

**Comment by changli6** [comment]
Thanks @jleach4 @dgomezferro suggestions .I'll modify the code.

**Comment by changli6** [comment]
Modified implementations and added column "Elapsed". Please review, thank you.

**Comment by yxia92** [comment]
Fix code has been approved in the master branch which is no different from branch-2.5, so merge it too.

**Pull request merged**